internals/search.py

```diff
@@ -35,6 +35,25 @@
 DEFAULT_RESULTS_PER_PAGE = 100
 
 
+def process_exclude_deleted_unlisted_query() -> Future:
+  """Return a future for all features, minus deleted and unlisted."""
+  query = FeatureEntry.query(
+      FeatureEntry.deleted == False,
+      FeatureEntry.unlisted == False)
+  future_feature_ids = query.fetch_async(keys_only=True)
+  return future_feature_ids
+
+
+def process_exclude_deleted_unlisted_enterprise_query() -> Future:
+  """Return a future for all features, minus deleted, unlisted, enterprise."""
+  query = FeatureEntry.query(
+      FeatureEntry.deleted == False,
+      FeatureEntry.unlisted == False,
+      FeatureEntry.feature_type <= core_enums.FEATURE_TYPE_DEPRECATION_ID)
```

Comment:
I don't know the answer to this, but could this be more efficient with the following change?

```suggestion
      FeatureEntry.feature_type != core_enums.FEATURE_TYPE_ENTERPRISE_ID)
```

Reply:
I don't think it would help.  My understanding is that NDB implements `!=` as `OR(query with <, query with >)`.  The second part of that `OR` should give an empty result, so it would be about the same.